Maths/Fraction.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @function fraction
+ * @description This function returns the fraction of a float type number.
+ * @param {number} num - a float type number is expected, but an integer will also work.
+ * @param {number} accuracy - the accuracy of the fraction, the default is 6.
+ * @return {Array} - an array containing the numerator and denominator of the fraction.
+ * @see https://en.wikipedia.org/wiki/Repeating_decimal and 
+ * @see https://en.wikipedia.org/wiki/Decimal#Decimal_fractions
+ * @example fraction(0.5) // [1, 2]
+ * @example fraction(0.3333333333333333) // [1, 3]
+ * @example fraction(0.25) // [1, 4]
+ * @example fraction(5.56) // [139, 25]
+ * @example fraction(0.33) // [33, 100]
+ */
+function fraction(number, accuracy = 6) {
+  let inp = typeof number === "number" ? number : false;
+  let acc = typeof accuracy === "number" ? accuracy : false;
+  if (inp !== false && acc !== false) {
+    if (Number.isInteger(inp)) {
+      return [inp, 1];
+    } else {
+      inp = inp.toString();
+      let len;
+      let reg = inp.match(/(\d+?)\1+$/);
+      if (reg && reg[0].length > acc) {
+        let pos = inp.split(".");
+        inp = inp.replace(reg[0], reg[1]);
+        let rec = pos[0] + pos[1].replace(reg[0], "");
+        inp = Number(rec + reg[1]) - Number(rec);
+        len = Number(
+          "9".repeat(reg[1].length) + "0".repeat(rec.length - pos[0].length)
+        );
+      } else {
+        inp = inp.replace(".", "");
+        len = 10 ** (inp.length - 1);
+        inp = Number(inp);
+      }
+      let d = true;
+      while (d) {
```

Comment:
This logic doesn't make much sense. It seems like a hack to handle multiples of 2, 3, 5, and 7 "better"? Either way this needs an explanation.

Reply:
Here I divide the numbers by some prime numbers every time, to get the smallest numerator and denominator. you can ask why only 2,3,5 and 7. Because, there is a unlimited count of prime numbers. In school, we learnt to divide using 2,3 and 5. Because that's how we get smallest numerator and denominator. I just added 7 here, nothing else.

Reply:
So the purpose of this is to shorten the fraction? If so, then this should simply divide both numerator and denominator by their GCD. No need to try out a set of primes.

Reply:
@appgurueu Yah! that's a great idea. should I add that, or you will!?